### DISCUSSION TITLE
How to get independent view when using NiceGui web deployment

### DISCUSSION THREAD

**Discussion by YH20220622** [zauberzeug/nicegui#915] [Q&A] [answered]
### Question

I tried to deploy the nicegui app on web, with:
`ui.run(host="192.168.3.18", port=9980)`

so, my local machine and another PC can visit the app through the browser at the same time. 
My problem is, these two machines have exactly the same view and same page contents. when I change something in one PC, the view will change in the other.

Is there anyway to get independent view？

Thanks for any help!

**Comment by rodja** [comment, marked as answer]
Use [`ui.page`](https://nicegui.io/documentation/page) to create private views for each visitor.

**Comment by rodja** [comment]
As a site note, NiceGUI allows access from external hosts by default. It uses the magic ip 0.0.0.0 which means "open on all interfaces". In our latest release 1.2.13, NiceGUI will also print all ip's on which the server is accessable.
